ecs/src/nox/ecs/createMetaInformation.tpp

```diff
@@ -10,115 +101,146 @@ nox::ecs::createMetaInformation(const TypeIdentifier& typeIdentifier)
     {
-        auto begin = static_cast<T*>(first);
-        auto end = static_cast<T*>(last);
-
-        while (begin != end)
-        {
-            begin->receiveEntityEvent(event);
-            ++begin;
-        }
+        static_cast<T*>(component)->~T();
     };
 
-    const auto moveConstruct = [](Component* dest, Component* src)
+    info.initialize =
+        getOperation<decltype(&Component::initialize),
+                     decltype(&T::initialize),
+                     decltype(info.initialize)>(
+                     [](Component* component,
+                        const Json::Value& value)
+                     {
+                         static_cast<T*>(component)->initialize(value);
+                     });
+
+    info.awake =
+        getOperation<decltype(&Component::awake),
+                     decltype(&T::awake),
+                     decltype(info.awake)>(
+                     [](Component* component)
+                     {
+                         static_cast<T*>(component)->awake();
+                     });
+
+    info.activate =
+        getOperation<decltype(&Component::activate),
+                     decltype(&T::activate),
+                     decltype(info.activate)>(
+                     [](Component* component)
+                     {
+                         static_cast<T*>(component)->activate();
+                     });
+
+    info.deactivate =
+        getOperation<decltype(&Component::deactivate),
+                     decltype(&T::deactivate),
+                     decltype(info.deactivate)>(
+                     [](Component* component)
+                     {
+                         static_cast<T*>(component)->deactivate();
+                     });
+
+    info.hibernate =
+        getOperation<decltype(&Component::hibernate),
+                     decltype(&T::hibernate),
+                     decltype(info.hibernate)>(
+                     [](Component* component)
+                     {
+                         static_cast<T*>(component)->hibernate();
+                     });
+
+    info.update =
+        getOperation<decltype(&Component::update),
+                     decltype(&T::update),
+                     decltype(info.update)>(
+                     [](Component* first,
+                        Component* last,
+                        const nox::Duration& deltaTime)
+                     {
+                         auto begin = static_cast<T*>(first);
+                         auto end = static_cast<T*>(last);
+
+                         while (begin != end)
+                         {
+                             begin->update(deltaTime);
+                             ++begin;
+                         }
+                     });
+
+    info.updateAccess = (info.update) ? info.updateAccess :
+                                        DataAccess::INDEPENDENT;
+
+    info.receiveLogicEvent =
+        getOperation<decltype(&Component::receiveLogicEvent),
```

Comment:
With this many template parameters, I might think that putting each parameter on each line in the same way you usually do it with arrays, would make it looks nicer, so like this:

getOperation
<
    decltype(&Component::receiveLogicEvent),
    decltype(&T::receiveLogicEvent),
    decltype(info.receiveLogicEvent)